RLBotDiscordBot/bot.py

```diff
@@ -46,15 +48,19 @@ def __init__(self):
     async def on_ready(self):
         self.logger.info(f'{self.user} online! (ID: {self.user.id})')
         self.has_reacted = 0
-        self.has_checked = False
 
     async def on_message(self, message):
+
         if not self.has_checked:
             for member in message.guild.members:
                 y = member.roles
                 for role in y:
                     if role.name == "Python" or role.name == "Java" or role.name == "C#" or role.name == "Rust" or role.name == "Scratch" or role.name == "C++":
-                        await member.add_roles(get(member.guild.roles, name="botmaker"), reason=None, atomic=True)
+                        if get(member.guild.roles, name="BotMaker") == None:
+                            self.has_checked = True
+                        else:
+                            await member.add_roles(get(member.guild.roles, name="BotMaker"), reason=None, atomic=True)
+                            self.has_checked = True
```

Comment:
This shouldn't need to be set on every member being checked. It should be set after the entire thing runs. This is a band-aid so that it doesn't run multiple times but it also slows things down for no reason.

Reply:
Also if you have the same piece of code in both if and the else you can put it outside the if statement. In this case you should invert the if statement and not have an else.